src/pkg/services/m365/api/groups.go

```diff
@@ -106,14 +110,44 @@ func (c Groups) GetByID(
 		return nil, err
 	}
 
-	resp, err := service.Client().Groups().ByGroupIdString(identifier).Get(ctx, nil)
-	if err != nil {
-		err := graph.Wrap(ctx, err, "getting group by id")
+	ctx = clues.Add(ctx, "resource_identifier", identifier)
 
-		return nil, err
+	var group models.Groupable
+
+	if uuidRE.MatchString(identifier) {
```

Comment:
A very corner case- if `uuid` is itself used a display name, the code will think it's a group ID and fail finding the correct group.

Reply:
Hmm, I guess it is unlikely. I guess we could look for names if the uuid look up fails in case we are really concerned about it.

Reply:
No harm in attempting the fallback, I suppose.